python/cudf/tests/test_sorting.py

```diff
@@ -166,3 +167,50 @@ def test_dataframe_nsmallest_sliced(counts, sliceobj):
     gdf = DataFrame.from_pandas(df)
     got = gdf[sliceobj].nsmallest(n, 'a')
     assert (got.to_pandas() == expect).all().all()
+
+
+@pytest.mark.parametrize('num_cols', [1, 2, 3, 5])
+@pytest.mark.parametrize('num_rows', [0, 1, 2, 1000])
+@pytest.mark.parametrize(
+    'dtype',
+    ['int8', 'int16', 'int32', 'int64', 'float32', 'float64',
+     'datetime64[ms]']
+)
+@pytest.mark.parametrize('nulls', ['none', 'some', 'all'])
+@pytest.mark.parametrize('ascending', [True, False])
+@pytest.mark.parametrize('na_position', ['first', 'last'])
+def test_dataframe_multi_column(num_cols, num_rows, dtype, nulls, ascending,
+                                na_position):
+    if dtype not in ['float32', 'float64'] and nulls in ['some', 'all']:
+        pytest.skip(msg='nulls not supported in dtype: ' + dtype)
+    if num_cols == 1 and nulls in ['some', 'all']:
+        pytest.skip(msg='single column sorting with nulls is '
+                        'non-deterministic for multi column dataframes')
+
+    from string import ascii_lowercase
+    np.random.seed(0)
+    by = list(ascii_lowercase[:num_cols])
+    pdf = pd.DataFrame()
+
+    for i in range(5):
```

Comment:
Think you want `for i in range(num_cols)`.

Reply:
No, I want to always generate 5 columns and test sorting on a different number of columns in the dataframe to ensure it works as expected. The problem is there's no easy way to check for correctness in the non-sort key columns as multiple orders would be valid based on the duplicate sort keys.